string operating_mode
string current_fsm